core/services/assets/AssetManager.php

```diff
@@ -123,6 +125,33 @@ public function addJavascript(
     }
 
 
+    /**
+     * @param string $handle
+     * @param array  $dependencies
+     * @param bool   $load_in_footer
+     * @return JavascriptAsset
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     * @throws DomainException
+     * @since $VID:$
+     */
+    public function addVendorJavascript(
+        $handle,
+        array $dependencies = array(),
+        $load_in_footer = true
+    ) {
+        $dev_suffix = wp_scripts_get_suffix('dev');
+        $vendor_path = $this->domain->pluginUrl() . 'assets/vendor/';
```

Comment:
should have mentioned this in my original comment, but if you think there is any chance that we will need to recreate the vendor path outside of this method, then this could be moved to its own method, ie:

```php
public function getVendorSourcePathForHandle($handle) {
    $dev_suffix = wp_scripts_get_suffix('dev');
    $vendor_path = $this->domain->pluginUrl() . 'assets/vendor/';
    return  "{$vendor_path}{$handle}{$dev_suffix}.js";
}
```

I can't immediately see how that would be necessary, so I'm fine with it as is, but if you think that could be useful, then better to do this now.

Reply:
ya I didn't create it outside of here (i.e. put it in the domain class) because I think its pretty specific to this need. I don't anticipate seeing more use-case for this and if we do it can be iterated on.